### ISSUE TITLE
How to copy fields with same name but different types?

### ISSUE THREAD

**Issue by ycydsxy** [jinzhu/copier#104]
## Your Question

I'd like to copy one field from a struct to another, but they are of different types, such as: 

```go
type User struct {
    Age int
}

type User2 struct {
    Age string // the field name is the same as User, but its type is different
}
```
As I know, the `copier` will skip this field in the copy process.

## Expected answer

 I'd like to define a custom parse function per field for the `copier` to invoke when it found fields with same name but different types.

**Comment by ThreadedStream** [comment]
In case when destination is an instance of User2 struct, Age (string) is assigned to a corresponding unicode character(garbage for large integer values), whereas ```copier``` won't do any copying in the opposite case.

**Comment by ycydsxy** [comment]
That's right. I know the current behavior of the `copier`, I am just wondering if a transformer is available in this situation. Maybe in this form:
```
type User2 struct {
    Age string `copier:"transform:SetAge"`
}

func (u *User2) SetAge(age int) {
    u.Age = fmt.Sprintf("%d", age)
}
```

**Comment by ThreadedStream** [comment]
Seems quite interesting. But i guess that the following syntax would be more appropriate
```go
type User2 struct {
      Age string `copier:"transform:int"`
}
```
